src/game/AI/ScriptDevAI/scripts/northrend/ulduar/ulduar/boss_ignis.cpp

```diff
@@ -334,48 +240,144 @@ bool EffectScriptEffectCreature_npc_iron_construct(Unit* pCaster, uint32 uiSpell
 // TODO Remove this 'script' when combat can be proper prevented from core-side
 struct npc_scorchAI : public Scripted_NoMovementAI
```

Comment:
Do SetCombatMovement false in reset in favor of not using Scripted_NoMovementAI and use react state passive, you can then delete the 3 horrible overrides.

Reply:
And turn off melee for good measure.